sensors/bms/bms/freya_bms_node.py

```diff
@@ -0,0 +1,129 @@
+import rclpy
+from rclpy.node import Node
+import rclpy.logging
+from sensor_msgs.msg import BatteryState
+from diagnostic_msgs.msg import DiagnosticStatus, KeyValue, DiagnosticArray
+from bms.freya_bms import BMS
+
+class FreyaBMSNode(Node):
+    """
+        Publishes Freya's BMS data to ROS2.
+
+        Methods:
+        ---------
+        publish_bms_data(self) -> None
+            publises BMS data from BMS system to ROS2 node.
+        create_key_value_pair(key: str, value) -> KeyValue:
+            creates KeyValue object from supplied key and value
```

Comment:
What does the KeyValue object part of the code do in this node, does it serve a purpose? Is it somehow published to diagnostics or anywhere else?
I need context, me no understand what that part of code end goal is lol

Reply:
The KeyValue object is needed for publishing to the /diagnostics topic. Basically the /diagnostics topic takes a message of type DiagnosticArray, which is an array of DiagnosticStatus messages. The DiagnosticStatus message contains a field  called values, which stores key-value pairs using the KeyValue object, so we can represent stuff like Voltage: 22.0 and stuff